src/Microsoft.Health.Checkpoints/Storage/BaseStorageCheckpointClient.cs

```diff
@@ -0,0 +1,90 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using EnsureThat;
+
+namespace Microsoft.Health.Checkpoints.Storage
+{
+    public class BaseStorageCheckpointClient : ICheckpointClient
+    {
+        private BlobContainerClient _storageClient;
+
+        public BaseStorageCheckpointClient(BlobContainerClient containerClient)
+        {
+            EnsureArg.IsNotNull(containerClient);
+
+            _storageClient = containerClient;
+        }
+
+        public virtual Task<ICheckpoint> GetCheckpointAsync(string checkpointIdentifier)
```

Comment:
Nit: Can this be null? If yes, perhaps add a default value `checkpointIdentifier = null`. If not add `EnsureArg.IsNotNullOrEmpty()`

Reply:
yep, thanks. updated to use EnsureArg.IsNotNullOrEmpty()